The requested pixel format 'W10 ' is not natively supported by the Linux kernel and likely requires a patch!

| Required Info                         |                                                                |
|---------------------------------|------------------------------------------- |
| Camera Model                       | {  D400 } | 
| Firmware Version                   | (Open RealSense Viewer --> Click info) | 
| Operating System & Version |    Linux (Ubuntu 16.04)   | 
| Kernel Version (Linux Only)    |   4.15.0-60                                         | 
| Platform                                 | PC|
| SDK Version                            |                        | 
| Language                            |  {C/C#/}                          | 
| Segment			|  {Robot }                          | 

### Issue Description

Hello I hava a D415 Depth Camera and it won't connect to my laptop.
The main error for this is mentioned below.
I can access the realsense viewer without any problems but when ever I turn my node on by using
roslaunch realsense2_camera rs_camera.launch
it gives this error. Please help me thank you

My kernel version is 4.15.0-60-generic

An exception has been thrown: The requested pixel format 'W10 ' is not natively supported by the Linux kernel and likely requires a patch!
Alternatively please upgrade to kernel 4.12 or later. Last Error: Invalid argument
terminate called after throwing an instance of 'rs2::backend_error'
what(): The requested pixel format 'W10 ' is not natively supported by the Linux kernel and likely requires a patch!
Alternatively please upgrade to kernel 4.12 or later. Last Error: Invalid argument
[camera/realsense2_camera_manager-1] process has died [pid 15139, exit code -6, cmd /opt/ros/kinetic/lib/nodelet/nodelet manager __name:=realsense2_camera_manager __log:=/home/hong/.ros/log/68c59866-ce8c-11e9-8e37-a86daaec59a0/camera-realsense2_camera_manager-1.log].
log file: /home/hong/.ros/log/68c59866-ce8c-11e9-8e37-a86daaec59a0/camera-realsense2_camera_manager-1*.log

@doronhi I'm facing a similar issue, with:
* SDK version 2.25.0-0~realsense0.1362
* Realsense ROS version 2.2.8
* Ubuntu 16.04
* Kernel: 4.4.0-131-generic
* Nvidia driver version: 390.48
* DKMS version: 1.3.4-0ubuntu1

I'm using nvidia gtx 1080. I installed everything from apt packages, and I see the following error:

```
A 2019-09-13T07:56:02.296917635Z  13/09 07:56:02,296 WARNING [140034679183104] (backend-v4l2.cpp:1149) Pixel format 20303157-0000-0010-8000-00aa003 likely requires patch for fourcc code W10 ! 
A 2019-09-13T07:56:02.298548176Z  13/09 07:56:02,298 ERROR [140034679183104] (types.h:243) The requested pixel format 'W10 ' is not natively supported by the Linux kernel and likely requires a patch! 
A 2019-09-13T07:56:02.298550162Z Alternatively please upgrade to kernel 4.12 or later. Last Error: Invalid argument 
A 2019-09-13T07:56:02.299257939Z  [31m[ERROR] [1568361362.299137722]: An exception has been thrown: The requested pixel format 'W10 ' is not natively supported by the Linux kernel and likely requires a patch! 
A 2019-09-13T07:56:02.299259868Z Alternatively please upgrade to kernel 4.12 or later. Last Error: Invalid argument [0m 
A 2019-09-13T07:56:02.299416264Z terminate called after throwing an instance of 'rs2::backend_error' 
A 2019-09-13T07:56:02.299436137Z   what():  The requested pixel format 'W10 ' is not natively supported by the Linux kernel and likely requires a patch! 
A 2019-09-13T07:56:02.299436648Z Alternatively please upgrade to kernel 4.12 or later. Last Error: Invalid argument 
A 2019-09-13T07:56:02.486309146Z  [31m[camera_cart_left/realsense2_camera_manager-1] process has died [pid 104, exit code -6, cmd /opt/ros/kinetic/lib/nodelet/nodelet manager __name:=realsense2_camera_manager __log:=/root/.ros/log/bbbf4e56-d5fb-11e9-b5e1-a0369f04326f/camera_cart_left-realsense2_camera_manager-1.log]. 
A 2019-09-13T07:56:02.486311496Z log file: /root/.ros/log/bbbf4e56-d5fb-11e9-b5e1-a0369f04326f/camera_cart_left-realsense2_camera_manager-1*.log [0m 
A 2019-09-13T07:56:02.975468492Z  [1mprocess[box_placing_depth_baseline_srv-1]: started with pid [438] [0m 
A 2019-09-13T07:56:02.976452294Z  [1mprocess[camera_calibration_downloader_eddie_333_1560272407429094588-1]: started with pid [444] [0m 
A 2019-09-13T07:56:02.976766641Z  [1mprocess[robot_state_publisher-1]: started with pid [447] [0m 
A 2019-09-13T07:56:03.077657230Z shutting down processing monitor... 
A 2019-09-13T07:56:03.077775899Z ... shutting down processing monitor complete 
A 2019-09-13T07:56:03.077818606Z  [1mdone [0m 
A 2019-09-13T07:56:03.200225446Z  [34mSuccessful exit from subprocess: ['roslaunch', 'nomagic_launch', 'camera_calibration_downloader.launch', 'camera_id:=box_placing', 'allow_no_internet:=False', 'camera_serial_no:=810512061758'] [0m 
```

What is interesting I don't see this error on another system with the same spec (or at least I failed to find differences).

Any advice?

--- a/realsense2_camera/include/base_realsense_node.h
+++ b/realsense2_camera/include/base_realsense_node.h
@@ -272,6 +272,7 @@ namespace realsense2_camera
         std::map<stream_index_pair, int> _width;
         std::map<stream_index_pair, int> _height;
         std::map<stream_index_pair, int> _fps;
+        std::map<rs2_stream, int>        _format;
         std::map<stream_index_pair, bool> _enable;
         std::map<rs2_stream, std::string> _stream_name;
         tf2_ros::StaticTransformBroadcaster _static_tf_broadcaster;

--- a/realsense2_camera/src/base_realsense_node.cpp
+++ b/realsense2_camera/src/base_realsense_node.cpp
@@ -96,6 +96,8 @@ BaseRealSenseNode::BaseRealSenseNode(ros::NodeHandle& nodeHandle,
     _depth_aligned_encoding[RS2_STREAM_DEPTH] = sensor_msgs::image_encodings::TYPE_16UC1;
 
     // Infrared stream
+    _format[RS2_STREAM_INFRARED] = RS2_FORMAT_Y8;
+
     _image_format[RS2_STREAM_INFRARED] = CV_8UC1;    // CVBridge type
     _encoding[RS2_STREAM_INFRARED] = sensor_msgs::image_encodings::MONO8; // ROS message type
     _unit_step_size[RS2_STREAM_INFRARED] = sizeof(uint8_t); // sensor_msgs::ImagePtr row step size
@@ -912,6 +914,7 @@ void BaseRealSenseNode::enable_devices()
                     (_width[elem] == 0 || video_profile.width() == _width[elem]) &&
                     (_height[elem] == 0 || video_profile.height() == _height[elem]) &&
                     (_fps[elem] == 0 || video_profile.fps() == _fps[elem]) &&
+                    (_format.find(elem.first) == _format.end() || video_profile.format() == _format[elem.first] ) &&
                     video_profile.stream_index() == elem.second)
                 {
                     _width[elem] = video_profile.width();
@@ -922,7 +925,7 @@ void BaseRealSenseNode::enable_devices()
 
                     _image[elem] = cv::Mat(_height[elem], _width[elem], _image_format[elem.first], cv::Scalar(0, 0, 0));
 
-                    ROS_INFO_STREAM(STREAM_NAME(elem) << " stream is enabled - width: " << _width[elem] << ", height: " << _height[elem] << ", fps: " << _fps[elem]);
+                    ROS_INFO_STREAM(STREAM_NAME(elem) << " stream is enabled - width: " << _width[elem] << ", height: " << _height[elem] << ", fps: " << _fps[elem] << ", " << "Format: " << video_profile.format());
                     break;
                 }
             }

